### DISCUSSION TITLE
Navigation dark mode quick option

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#126] [Q&A] [answered]
During a recent long journey, I found myself driving into the evening where it got dark - and my "light" theme became an annoyance so I had to pull over and switch to my "dark" map theme. It occurs to me that there is also the scenario of driving through a long tunnel or other times where you wish to use a "dark" map theme without too much phone-faffing (ie: safe operation) - and this kind of switch is often taken care of by the phone's light sensor.

[Here's an example of how CoPilot GPS offers this feature.](https://support.copilotgps.com/en/support/solutions/articles/19000046325-how-do-i-modify-brightness-and-or-the-color-and-style-of-my-map-)

How easy would it be to have a setting for light+dark situations where as soon as you cross the light threshold, the theme swaps over? Perhaps not as fancy as the above example, but a "picker" like the existing theme dialog that you can invoke for light+dark preferences?

**Comment by devemux86** [comment]
@MotoUKRider thanks for the suggestion.

How many users actually use dark themes and find them useful in navigation?
Isn't it simpler to have a profile with dark theme and change it when circumstances require it?

---

Options could be:
- User
- Sunrise / sunset
- Day
- Night
- Light sensor

**Reply by MotoUKRider** [reply]
For me, it was the bright map colours that were the issue - but perhaps there is scope to include the app's overall GUI to match the "surroundings"?  I'm just thinking from a safety aspect that you may not like the glare of a bright map, and not have the ability (or opportunity) to safely change it.

I suppose we need to decide where this setting goes because Map Theme and App Theme are in 2 separate places :)

**Comment by devemux86** [comment]
It could be more things:
- App theme
- Map theme
- Interface colors: Display, Icon(s), Font
- Route color
- Location color
- ...

The possibilities are many, so it could become complicated.

**Reply by MotoUKRider** [reply]
Indeed, for example, the blue system colour of the icons doesn't look good in "Dark" app mode. There's so much to switch around, as I found out when converting my map theme into a dark one!

One other aspect that may allow for a more easier (map) theme swap: The labelling of the ZIP files is included, is there any way to make this *just* the theme XML name?

eg: `style-motorrideruk.zip/dark-mode` becomes `dark-mode` or whatever the selected XML file within the ZIP is called.  Obviously you lose the reference name of the parent theme (like Elevate, Mapsforge or whoever), but it reads easier on the small dialog window.

**Reply by devemux86** [reply]
> the blue system colour of the icons doesn't look good in "Dark" app mode

What icons do you mean?

> The labelling of the ZIP files is included, is there any way to make this just the theme XML name?

Thanks for the report, I will check it!

**Reply by MotoUKRider** [reply]
The main menu of large icons: Map, Routing, Favorites etc (set to 0000FF).  The Settings menu looks better as they adopt a more cyan colour to the icons and revert to Cruiser-blue when in "light" app mode.

This is maybe because you can customise the icon colour for the on-screen "large" buttons but the Settings menu seems to be an internal configuration between Cruiser-blue and Cyan-blue?

**Reply by devemux86** [reply]
> the Settings menu seems to be an internal configuration between Cruiser-blue and Cyan-blue?

That's right, it's the app's predefined accent color, Android doesn't seem to allow changes to it.
It is used automatically in Android's default screens / dialogs.

> This is maybe because you can customise the icon colour for the on-screen "large" buttons

These are custom screens, so we can offer color option to the users.
(or use the predefined blue / cyan color combination?)

**Reply by devemux86** [reply]
> The labelling of the ZIP files is included, is there any way to make this just the theme XML name?

This will be available in the next Cruiser update.

**Comment by MotoUKRider** [comment]
This'll probably find its way into the Wiki eventually, but my current solution to comfortable/safe night navigation is to use a 2nd profile rather than just change the map theme.  It allows me to tailor the screen better.

Click the 3 dots on your existing profile and make a new one, using these settings as the "base":

![cruiser-night-profile](https://github.com/devemux86/cruiser/assets/125696109/677df952-58bf-4c4f-b895-abe1bdf35fba)

Then, change the Map Theme to "Motorider Dark" and set your App/Panel/Text colours to suit.

Day/Night:
![cruiser-navigation-day-and-night-s](https://github.com/user-attachments/assets/60a44af0-7e0a-47a3-ba15-cc7bc3e262e1)

*Edit:* This post was updated with a new Day/Night screenshot.

**Reply by devemux86** [reply]
@MotoUKRider thanks for the very useful solution!

You could add this to the [Wiki](https://github.com/devemux86/cruiser/wiki) if you want. :slightly_smiling_face:

**Reply by MotoUKRider** [reply]
@devemux86 Added! [Profile menu](https://github.com/devemux86/cruiser/wiki/Profile-menu) page updated.

**Reply by devemux86** [reply]
@MotoUKRider thanks, very nice work! :+1:

**Comment by WolfgangZ42** [comment]
Since my screenshots don't live longer than a few minutes - here is just a screenshot of my screenshots  🥴

![image](https://github.com/user-attachments/assets/ab4aed00-9493-4cbb-93b7-23abea5175b9)

**Reply by devemux86** [reply]
@WolfgangZ42 You can edit the other post and copy the text + image links here.

**Comment by Maverikk65** [comment]
Hello, I switched liked @WolfgangZ42 proposed  but to add a profile seems to be the better solution... Especially in the night I have problems with some colors (like @MotoUKRider mentioned) and I would like to have a good view on the route and to have good visibility of the roads I might need in case of a re-routing... This is very difficult on many navigations, to clearly see those streets (autobahn, roads, either too many details or too few) in the night... 
Actually in Copliot there was a nice feature to switch the day/night/auto mode by 2 clicks (in the very last version before they broke it). 
As a car driver I need the night feature very often... 
Danke, VG, NG.

**Reply by devemux86** [reply]
The default (light / dark) map theme in Cruiser is written by @MotoUKRider.
We can always improve the map theme appearance and colors.

@Maverikk65 
If you have any suggestions for improvements, please write them in the map theme thread:
- https://github.com/devemux86/cruiser/discussions/612

**Comment by airald** [comment]
I also I'm using profiles since exists to switch to night mode. Before I just changed map theme, but also is better for me change colors on the app.

**Comment by bios4** [comment]
What about adding sort of a "quick action" button, accessible during naviagtion e.g. in the menu which is opened by tapping the naviagtion arrow once!?

New tile --> switch to night mode --> selecting it activates a "dark map" or "night mode" with ONE click?

Of course this "night mode" would need to be setup ONCE, e.g. by defining in the profile what map theme (or color theme) should be used in this "night mode".

Would this be feasible to implement @devemux86?

**Reply by devemux86** [reply]
Using external theme combinations (light/dark) makes it much more complicated to handle.
Also there are not many map themes with a dark version, other than our default map theme.

**Reply by devemux86** [reply]
> What about adding sort of a "quick action" button, accessible during naviagtion e.g. in the menu which is opened by tapping the naviagtion arrow once!?
New tile --> switch to night mode --> selecting it activates a "dark map" or "night mode" with ONE click?

This is a nice idea, a quick alternative to a possible automatic switch (discussed [above](https://github.com/devemux86/cruiser/discussions/126#discussioncomment-5579422)).

**Comment by devemux86** [comment]
Dark mode quick option in navigation will be available in **Cruiser 5.2**.
(long press the direction icon at top left)

It works with the internal light / dark map themes.

<img src="https://github.com/user-attachments/assets/ca585af0-1dc4-4077-96e9-fc1ad155a3b0" width="50%"/><img src="https://github.com/user-attachments/assets/47ccbad4-a338-4527-a824-b0f9fc97e1c4" width="50%"/>

**Reply by bios4** [reply]
Very nice! 👍

May I suggest to really call it just "Night mode" (or ""Dark mode") and use the `dark_mode` material icon for the menu tile!??

**Reply by bios4** [reply]
> It currently works with the default light / dark map themes.

Could it be implemented in a way that it works with any map style - which is defined by the user?

**Reply by devemux86** [reply]
@bios4 These were navigation **actions**, i.e. a button that alternates between light and dark theme.

Or we could move it to navigation **options**? (long press direction icon at top left)
Then it could be an on/off switch with on/off status and call it by another name.

**Reply by devemux86** [reply]
> Could it be implemented in a way that it works with any map style - which is defined by the user?

This would be complicated, we'll see in the future. So currently it works with the default map themes.
Note that only the Motorider theme has a dark version, not the OpenAndroMaps themes.

Also it only changes the map theme. App appearance can only be changed in app settings.

**Reply by bios4** [reply]
> These are navigation actions, i.e. a button that alternates between light and dark theme.

Yes, I understand that (that's what I suggested)... 😉

What I mean is:
- Define in (navigation; app; ...) options which map style/app theme is meant to be used for the "Night mode" (i.e. the default "Motorider dark").
- The action after tapping the navigation arrow > "Night mode" button then "executes" the switch between the default and the dark theme (and vice versa).

**Reply by devemux86** [reply]
Dark mode will be a quick option in navigation with on / off status.
(long press the direction icon at top left)

I updated the screenshot above:
- https://github.com/devemux86/cruiser/discussions/126#discussioncomment-11460963

**Reply by DolomitesTom** [reply]
will the current map layer settings be kept? Because before when you changed the map style, the map layers were always set back to initial.

**Reply by devemux86** [reply]
From now on, when we switch between the default light and dark theme, the map layers are preserved.

**Comment by devemux86** [comment]
Available for testing in:
- https://github.com/devemux86/cruiser/discussions/690

It works with the internal light / dark map themes.
(long press the direction icon at top left)

**Reply by bios4** [reply]
Switching works flawlessly, thank you! 👍

> long press the direction icon at top left

Is there also a possibility to switch to "night mode" WITHOUT having a navigation running?
E.g. for (re)planning in the night?

**Reply by devemux86** [reply]
> Is there also a possibility to switch to "night mode" WITHOUT having a navigation running?
E.g. for (re)planning in the night?

This is available in the menu "Map | Map theme".

**Reply by bios4** [reply]
So just the default way of switching themes, and the answer is "no, there is no quick-switch action available for that out of navigation".
Alright.

Thanks!

**Comment by DolomitesTom** [comment]
In dark mode, the route color remains the same. As a result, the contrast is poor. It should also be possible to define a different color for the dark mode. And how can you remove the pattern structure in wooded areas?

**Reply by devemux86** [reply]
Dark mode quick option can only change the map theme.

You can choose a route color that works on both light and dark themes., e.g. the default magenta color.

Or use the app profiles to switch light / dark themes as demonstrated above by @MotoUKRider.
Then you can customize all app appearance options.
- https://github.com/devemux86/cruiser/discussions/126#discussioncomment-9568315

**Comment by devemux86** [comment, marked as answer]
- Navigation dark mode quick option is available in https://github.com/devemux86/cruiser/discussions/701
